README.md

```diff
@@ -139,7 +139,7 @@ function afterPrerendering() {
 if (!document.loadingMode || document.loadingMode.type === 'default') {
   afterPrerendering();
 } else {
-  document.addEventListener('loadingmodechange', () => {
+  document.loadingMode.addEventListener('change', () => {
```

Comment:
This has the knock-on effect of requiring this to be an interface so that it can derive EventTarget, rather than being a dictionary, and also implies that its set of properties must remain fixed since the properties of a platform object don't change and we cannot swap it out for a different one because it has listeners.

That's why semantically I thought of it as changes to a property of the document, as opposed to an active thing in and of itself (analogous to orientationchange, languagechange, readystatechange, paymentmethodchange, etc).

Reply:
It can't be a dictionary though, since it's returned from a Web IDL attribute. If it were a dictionary we'd need `document.loadingMode()` or similar.

Is there a situation in which we'd expect the properties to change? Stated another way, do you have a vision for this API that goes beyond what's outlined in the API section here?

Reply:
Hmm, I didn't realize IDL attributes can't return dictionaries. Basically I'm wondering if we'll want to parameterize these in the future and tell the content things like `{type: 'isolated-prerender', allowedOrigins: [...]}` or `{type: 'uncredentialed-prerender', viaProxy: '...'}`. I don't have concrete use cases for those at the moment, but if it's not too far out of the way it would be nice to allow for it in the future.

I suppose we do always have the option of later changing to a union of some kind but that's still potentially breaking.

I might be overengineering here. If so, maybe this should just be an enum and a `loadingmodechange` event. If not...hmm, what else could we do? If we do go for a platform object (like an `EventTarget`), it seems mildly weird to me to call it `loadingMode` (which sounds like inert data to me).

Reply:
I feel like we have three paths for this API:

- Super-minimal; no dedicated API. Maybe add a `"prerender"` mode to `visibilityState`, since per @bokan that seems web-compatible. (And it already has a change event.) But mostly people rely on purpose-specific APIs like permissions and storage access.

- Minimal but extensible. That's what my proposal here is trying to do: it has a `LoadingMode` interface, which so far only has `EventTarget` / `onchange` and the `type` property, but could grow if we want it to.

- Maximal, i.e. try to figure out the whole API including parameterization before shipping.

Maybe super-minimal is best?

Reply:
I like starting with super-minimal, but getting back to this one once we started to feel something like this may be needed could probably work?

Reply:
I'll make it clear this is a more on-going discussion. We can centralize that discussion in #2.